### PR TITLE
Business Hours Block: Site Editor Compatibility:  editing styles

### LINKED ISSUE #19463: Site Editor Compatibility: Business Hours Block styling issue with open/closed toggle
In the new site editor in Gutenberg (/wp-admin/admin.php?page=gutenberg-edit-site), there are some styling issues with the Business Hours block when rendered in the site editor.

#### Steps to reproduce the issue

Related to: 256-gh-Automattic/view-design

1. With the Gutenberg plugin active (>=10.4.0-rc1), activate a block-ready theme (like tt1-blocks)
2. In the site editor, insert a Business Hours block into the Footer template part.
3. Observe that the size / placement of the Open / Closed button differs to when rendered in the post/page editor (the CSS-in-JS styles that affect the font size don't appear to be carried over, as in https://github.com/WordPress/gutenberg/issues/30553)

#### What I expected

The UI in the editor canvas to look the same as in the post/page editor

#### What happened instead
The font size of the controls is slightly bigger, and the Open / Closed toggle is a bit lower than it should be.

**Screenshots**

| FSE running tt1-blocks theme | Post editor in Twenty Twenty One theme |
| --- | --- |
| image | image |

### PR DESCRIPTION
**Pull request by ramonjd**
Fixes #19463

#### Changes proposed in this Pull Request:
In the new site editor in Gutenberg (/wp-admin/admin.php?page=gutenberg-edit-site), there are some styling issues with the Business Hours block when rendered in the site editor.

Most notably font sizes, padding and toggle switch disco positioning. 🕺 

From what I can tell the Site Editor is:
1. Not pulling in the common styles from the theme into the iframe.
2. Resetting the styles of a few controls,

This PR attempt to make more specific CSS declarations in the block itself so that the editing mode version of the block look kinda the same across the two editors.

Screen Shot 2021-04-14 at 3 58 38 pm

#### Jetpack product discussion
This is part of a [Jetpack FSE audit.](https://github.com/Automattic/view-design/issues/256)

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. With the Gutenberg plugin active (>=10.4.0-rc1), activate a block-ready theme (like tt1-blocks)
2. In the site editor, insert a Business Hours block into the Footer template part.
3. Click on the blocks to enter editing mode
4. Compare the block in the two editors. Do they look okay? No major differences? Have you brushed your teeth lately.?

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/business-hours/editor.scss [hunk 1]
```diff
@@ -36,7 +51,7 @@
 		.business-hours__day {
 			width: 44%;
 			display: flex;
-			align-items: baseline;
+			align-items: end;
```
Comment (andrewserong) [inline projects/plugins/jetpack/extensions/blocks/business-hours/editor.scss:54]: This is testing well for me in Chrome and Edge so far, but in Firefox setting to `align-items: end` results in the day text and toggle button being lower down within the element:

image

Swapping this out for `align-items: start` appears to fix it, though, and I think works the same in Chrome, Edge and Safari?

```suggestion
			align-items: start;
```

**Review by andrewserong — CHANGES_REQUESTED** [review]
Nice changes getting the overrides working here @ramonjd! It's testing well for me across Chrome and Edge, but I noticed an issue with `align-items: end` in FF. I _think_ if we switch it to `start`, it should fix it in each of them, if I'm not missing anything?

**Comment by ramonjd** [comment]
Thanks @andrewserong for testing and for the suggestion.
FF:
Screen Shot 2021-04-16 at 12 23 49 pm

Also double checked in other browsers. 👍

**Review by andrewserong — APPROVED** [review]
Looks good, thanks @ramonjd! That's testing nicely in FF, Safari, Chrome, and Edge for me now. 👍

It looks like CI is complaining that the PR isn't current, so this might just need a `git merge origin/master` to stop it from complaining.

**Review by samiff — APPROVED** [review]
LGTM 🚢

**Comment by ramonjd** [comment]
r224441-wpcom

**Pull request merged**